main.cpp

```diff
@@ -1 +1,239 @@
+    vector<string> list{"special", "random", "roast", "firetruck", "throw", "trend", "blast", "gasoline"}; // list of possible questions
+    vector<char> correctguesses;                                                                           // letters that were guessed correctly
+    int guesses;                                                                                           // number of guesses by the user
+    srand(time(NULL));
+    int random = rand() % list.size(); // randomise the word
+    string word = list[random];
+    printoptions(); // to print the options
+    // Start Code
+    while (guesses <= 10)
+    {
+        cout << "Remaining Guesses: " << 10 - guesses << endl;
+        print(guesses); // print hangman
+        cout << endl;
+        if (guesses == 10) // if user has lost the game
+        {
+            cout << "\nMission Failed. We'll get them next time." << endl;
+            cout << "\nThe word is " << word << "." << endl;
+            cout << "Would you like to play again? (Y/N)" << endl;
+            input = ' '; // resets the char 'input'
+            while (input != 'y' || input != 'Y')
+            {
+                cin >> input;
+                if (input == 'y' || input == 'Y')
+                {
+                    guesses = 0;
+                    guessedchar.clear();
+                    srand(time(NULL));
+                    int random = rand() % list.size(); // randomise the word
+                    string word = list[random];
+                    printoptions();
+                    break;
+                }
+                else if (input == 'n' || input == 'N')
+                {
+                    return 0;
+                }
+                else
+                {
+                    cout << "Invalid option please type either 'y' or 'n'" << endl;
+                }
+            }
+        }
+        else if (guesses < 10) // if user still has guesses remaining
+        {
+            cout << "The word is " << word.length() << " characters long." << endl;
+            printword(word, correctguesses); // print the question
+            cout << "\nYou have " << word.length() - correctguesses.size() << " letters remaining." << endl;
+            cout << "\nChoose your option: ";
+            cin >> choice;
+            while (cin.fail()) // check for valid input
+            {
+                cout << "Invalid Input. Please enter a valid number. Enter '4' to view options again" << endl;
+                cin.clear();
+                cin.ignore(numeric_limits<streamsize>::max(), '\n');
+                cin >> choice;
+            }
+            cout << "\nYou have chosen [" << choice << "]\n";
+        }
+        switch (choice)
+        {
+        case 1: // when user decides to guess a letter
+            cout << "Enter your letter: ";
+            cin >> input;
+            if (count(guessedchar.begin(), guessedchar.end(), input))
```

Comment:
Never seen this incredibly advanced check, i think a simpler `guessedchar.size() == input` is easier to read, but that's just me maybe